### PR TITLE
Camera plugin auto-generation

### PR DESCRIPTION
**Pull request by julianoes**

### CODE REVIEW

File: protos/camera/camera.proto [hunk 1]
```diff
@@ -38,45 +40,76 @@ service CameraService {
...
+    rpc SubscribeMode(SubscribeModeRequest) returns(stream ModeResponse) { option (mavsdk.options.async_type) = ASYNC; }
     /*
      * Subscribe to video stream info updates.
      */
-    rpc SubscribeVideoStreamInfo(SubscribeVideoStreamInfoRequest) returns(stream VideoStreamInfoResponse) {}
+    rpc SubscribeVideoStreamInfo(SubscribeVideoStreamInfoRequest) returns(stream VideoStreamInfoResponse) { option (mavsdk.options.async_type) = ASYNC; }
+    /*
+     * Get video stream information.
+     */
+    //rpc GetVideoStreamInfo(GetVideoStreamInfoRequest) returns(GetVideoStreamInfoResponse) { option (mavsdk.options.async_type) = SYNC; }
```
Comment (JonasVautherin) [inline protos/camera/camera.proto:67]:
```suggestion
```

I think we can just remove that.

File: protos/camera/camera.proto [hunk 2]
```diff
@@ -38,45 +40,76 @@ service CameraService {
...
+     */
+    rpc GetSetting(GetSettingRequest) returns(GetSettingResponse) {}
+    /*
+     * Get information about a camera.
+     */
+    //rpc GetInformation(GetInformationRequest) returns(GetInformationResponse) { option (mavsdk.options.async_type) = SYNC; }
+    /*
+     * Get status of a camera.
+     */
+    //rpc GetStatus(GetStatusRequest) returns(GetStatusResponse) { option (mavsdk.options.async_type) = SYNC; }
```
Comment (JonasVautherin) [inline protos/camera/camera.proto:103]:
```suggestion
```

Same here

File: protos/camera/camera.proto [hunk 3]
```diff
@@ -38,45 +40,76 @@ service CameraService {
...
+    /*
+     * Format storage (e.g. SD card) in camera.
+     *
+     * This will delete all content of the camera storage!
+     */
+     rpc FormatStorage(FormatStorageRequest) returns(FormatStorageResponse) {}
 }
 
+
```
Comment (JonasVautherin) [inline protos/camera/camera.proto:112]:
```suggestion
```

File: protos/camera/camera.proto [hunk 4]
```diff
@@ -115,30 +148,42 @@ message StopVideoStreamingResponse {
...
 message SetModeResponse {
     CameraResult camera_result = 1;
 }
 
+//message GetModeRequest {}
+//message GetModeResponse {
+//    CameraResult camera_result = 1;
+//    Mode mode = 2; // Camera mode
+//}
+
```
Comment (JonasVautherin) [inline protos/camera/camera.proto:162]:
```suggestion
```

File: protos/camera/camera.proto [hunk 5]
```diff
@@ -115,30 +148,42 @@ message StopVideoStreamingResponse {
...
 message VideoStreamInfoResponse {
     VideoStreamInfo video_stream_info = 1; // Video stream info
 }
 
+//message GetVideoStreamInfoRequest {}
+//message GetVideoStreamInfoResponse {
+//    CameraResult camera_result = 1;
+//    VideoStreamInfo video_stream_info = 2; // Video stream info
+//}
+
```
Comment (JonasVautherin) [inline protos/camera/camera.proto:178]:
```suggestion
```

File: protos/camera/camera.proto [hunk 6]
```diff
@@ -158,40 +203,54 @@ message SetSettingResponse {
...
+    CameraResult camera_result = 1;
+    Setting setting = 2; // Setting
+}
+
+//message GetInformationRequest {}
+//message GetInformationResponse {
+//    CameraResult camera_result = 1;
+//    Information information = 2; // The information
+//}
+
```
Comment (JonasVautherin) [inline protos/camera/camera.proto:219]:
```suggestion
```

File: protos/camera/camera.proto [hunk 7]
```diff
@@ -275,22 +346,30 @@ message CameraStatus {
...
     string setting_id = 1; // Name of a setting (machine readable)
     string setting_description = 2; // Description of the setting (human readable)
     Option option = 3; // Selected option
-}
-
-// Type to represent a setting option.
-message Option {
-    string option_id = 1; // Name of the option (machine readable)
-    string option_description = 2; // Description of the option (human readable)
+    bool is_range = 4; // If option is given as a range
```
Comment (JonasVautherin) [inline protos/camera/camera.proto:360]: So now this is like `setting_description`, right? I.e. when the user creates the object, they should ignore this field. Correct?

**Comment by julianoes** [comment]
I see, this needs cleanup :see_no_evil:

File: protos/camera/camera.proto [hunk 8]
```diff
@@ -38,45 +40,76 @@ service CameraService {
...
-    rpc StopVideoStreaming(StopVideoStreamingRequest) returns(StopVideoStreamingResponse) {}
+    rpc StopVideoStreaming(StopVideoStreamingRequest) returns(StopVideoStreamingResponse) { option (mavsdk.options.async_type) = SYNC; }
     /*
      * Set camera mode.
      */
     rpc SetMode(SetModeRequest) returns(SetModeResponse) {}
+    /*
+     * Get camera mode.
+     */
+    //rpc GetMode(GetModeRequest) returns(GetModeResponse) { option (mavsdk.options.async_type) = SYNC; }
```
Comment (JonasVautherin) [inline protos/camera/camera.proto:55]:
```suggestion
```

File: protos/camera/camera.proto [hunk 9]
```diff
@@ -49,30 +49,26 @@ service CameraService {
...
-     */
-    //rpc GetMode(GetModeRequest) returns(GetModeResponse) { option (mavsdk.options.async_type) = SYNC; }
     /*
      * Subscribe to camera mode updates.
      */
-    rpc SubscribeMode(SubscribeModeRequest) returns(stream ModeResponse) { option (mavsdk.options.async_type) = ASYNC; }
+    rpc SubscribeMode(SubscribeModeRequest) returns(stream ModeResponse) {}
     /*
-     * Subscribe to video stream info updates.
+     * Subscribe to camera information updates..
```
Comment (JonasVautherin) [inline protos/camera/camera.proto:57]:
```suggestion
     * Subscribe to camera information updates.
```

File: protos/camera/camera.proto (hunk 7 above)
Comment (JonasVautherin) [reply protos/camera/camera.proto:328]: What about I add a default value for them (`""` and `false`)? Good enough?

**Pull request merged**